dsf-fhir/dsf-fhir-server/src/test/java/org/highmed/dsf/fhir/integration/BinaryIntegrationTest.java

```diff
@@ -1613,6 +1613,43 @@ public void testDelete() throws Exception
 		binaryDao.read(UUID.fromString(created.getIdElement().getIdPart()));
 	}
 
+	@Test(expected = ResourceDeletedException.class)
+	public void testDeleteDeleted() throws Exception
+	{
+		OrganizationDao orgDao = getSpringWebApplicationContext().getBean(OrganizationDao.class);
+		Organization createdOrg = orgDao.create(new Organization());
+		
+		final String contentType = MediaType.TEXT_PLAIN;
+		final byte[] data1 = "Hello World".getBytes(StandardCharsets.UTF_8);
+		
+		Binary binary = new Binary();
+		binary.setContentType(contentType);
+		binary.setData(data1);
+		binary.setSecurityContext(new Reference(createdOrg.getIdElement().toVersionless()));
+		
+		BinaryDao binaryDao = getSpringWebApplicationContext().getBean(BinaryDao.class);
+		Binary created = binaryDao.create(binary);
+		
+		assertNotNull(created);
+		assertNotNull(created.getIdElement().toString());
+		assertEquals("1", created.getMeta().getVersionId());
+		assertNotNull(created.getMeta().getLastUpdated());
+		
+		assertNotNull(created.getContentType());
+		assertEquals(contentType, created.getContentType());
+		assertTrue(Arrays.equals(data1, created.getData()));
+		
+		assertNotNull(created.getSecurityContext());
+		assertEquals(createdOrg.getIdElement().toVersionless(), created.getSecurityContext().getReferenceElement());
+		
+		boolean deleted = binaryDao.delete(UUID.fromString(created.getIdElement().getIdPart()));
+		assertTrue(deleted);
+
+		getWebserviceClient().delete(Binary.class, created.getIdElement().getIdPart());
+		
+		binaryDao.read(UUID.fromString(created.getIdElement().getIdPart()));
```

Comment:
I think this line does not return and throw the expected exception.

Reply:
hanging test based on bug in fhir client, fix see 1c17efe